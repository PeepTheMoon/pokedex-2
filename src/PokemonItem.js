import React, { Component } from 'react'

export default class PokemonItem extends Component {
    render() {
        return (
            <ul>
                <li className="pokemon-item">
                    <h2 className="pokemon-name">{this.props.character.pokemon.toUpperCase()}</h2>
                    <img className="pokemon-character" alt={this.props.character.pokemon} src={this.props.character.url_image}/>
                    <p className="pokemon-type">Type = {this.props.character.type_1}</p>
                    <p className="pokemon-attack">Attack = {this.props.character.attack}</p>
                    <p className="pokemon-defense">Defense ={this.props.character.defense}</p>
                </li>
                
            </ul>
        )
    }
}
