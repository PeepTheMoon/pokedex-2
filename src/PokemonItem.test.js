import React from 'react';
import { shallow } from 'enzyme';
import PokemonItem from './PokemonItem.js';

it('renders pokemon item', () => {
    const wrapper = shallow(<PokemonItem character={{ pokemon: 'character',url_image: 'some image', type_1: 'some type', attack: 0, defense: 0}}/>);
  
    expect(wrapper).toMatchSnapshot();
  });