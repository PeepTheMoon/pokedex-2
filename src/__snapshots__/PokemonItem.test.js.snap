// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<ul>
  <li
    className="pokemon-item"
  >
    <h2
      className="pokemon-name"
    >
      CHARACTER
    </h2>
    <img
      alt="character"
      className="pokemon-character"
      src="some image"
    />
    <p
      className="pokemon-type"
    >
      Type = 
      some type
    </p>
    <p
      className="pokemon-attack"
    >
      Attack = 
      0
    </p>
    <p
      className="pokemon-defense"
    >
      Defense =
      0
    </p>
  </li>
</ul>
`;
